include/zio/zio_dev.h

```diff
@@ -0,0 +1,340 @@
+ * |                                                                         |
+ * | Device Attributes-----------------------------------------------------+ |
+ * | | +-----------------------------------------------------------------+ | |
+ * | | |                Device Attribute 0 (Ex: Op. Mode)                | | |
+ * | | +-----------------------------------------------------------------+ | |
+ * |                                                                         |
+ * | +---------------------------------------------------------------------+ |
+ * | Channels--------------------------------------------------------------+ |
+ * | | Channel 0---------------------------------------------------------+ | |
+ * | | | Channel Attributes---------------------------------------------+| | |
+ * | | | | +----------------------------------------------------------+ || | |
+ * | | | | |      Channel Attribute 0 (Ex: Sample Rate)               | || | |
+ * | | | | +----------------------------------------------------------+ || | |
+ * | | | | |      Channel Attribute 1 (Ex: Scale)                     | || | |
+ * | | | | +----------------------------------------------------------+ || | |
+ * | | | | |      Channel Attribute 2 (Ex: Low-Pass Filter Frequency) | || | |
+ * | | | | +----------------------------------------------------------+ || | |
+ * | | | +--------------------------------------------------------------+| | |
+ * | | +-----------------------------------------------------------------+ | |
+ * | | |Channel 1                                                        | | |
+ * | | +-----------------------------------------------------------------+ | |
+ * | | |Channel 2                                                        | | |
+ * | | +-----------------------------------------------------------------+ | |
+ * | +---------------------------------------------------------------------+ |
+ * +-------------------------------------------------------------------------+
+ *
+ * Each device describes available attributes and channels through a set of
+ * arrays containing description structs available by the zio_dev API. The
+ * device driver can and probably does define these statically, though they
+ * could be dynamic if needed.
+ *
+ * The interface expects attributes are infrequently read or manipulated where
+ * as datum reads or writes to the set of device channels to be very frequent.
+ *
+ * Manipulating attributes of a device is done by referencing the
+ * index of the attribute in the devices array of attribute descriptions.
+ *
+ * Channel attributes may also be manipulated in the same manner as device
+ * attributes.
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZIO_DEV_H_
+#define ZEPHYR_INCLUDE_ZIO_DEV_H_
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#include <zio/zio_attr.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief ZIO device API definitions.
+ * @defgroup zio_device ZIO device definitions
```

Comment:
We'll also need to add a .rst file that references this group name to have the zio_device documentation appear in our API docs.